catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -346,6 +329,40 @@ private Filter parseFilter(FilterType filterType) throws CswException {
         return (Filter) parseJaxB(filterElement);
     }
 
+    private QueryRequest transformQuery(QueryRequest request, List<QName> typeNames) {
+        QueryRequest result = request;
+        for (QName typeName : typeNames) {
+            QueryFilterTransformer transformer = queryFilterTransformerProvider.getTransformer(
+                    typeName);
+            if (transformer != null) {
```

Comment:
I was thinking more of something along those lines:
```
result = queryFilterTransformerProvider.getTransformer(typeName)
        .map(it -> it.transform(temp, null))
        .orElse(result);
```

Didn't try it but it _should_ work 😉